Unable to import 'collections.abc'  using python 3.13.1

### Bug description

```python
import collections.abc
```

### Configuration

```ini

```

### Command used

```shell
We use a complex github actions but notice the change when python updated to 3.13.1
using pylint-3.3.2
```

### Pylint output

```python
************* Module spinn_utilities.checker
spinn_utilities/checker.py:1:0: C0304: Final newline missing (missing-final-newline)
spinn_utilities/checker.py:1:0: E0401: Unable to import 'collections.abc' (import-error)
spinn_utilities/checker.py:1:0: W0611: Unused import collections.abc (unused-import)
```

### Expected behavior

Not seeing the Unable to import 'collections.abc' (import-error)

Other messages are due to the tiny example

likely due to
https://github.com/python/cpython/commit/d46d291bd390027811937bbf5af2e2600b8398a9

### Pylint version

```shell
Python 3.13.1

Did not happen with 
Python 3.13.0
```

### OS / Environment

tested with github actions using 
Ubuntu
  22.0[4](https://github.com/SpiNNakerManchester/SpiNNUtils/actions/runs/12199055579/job/34032094669#step:1:4).5
  LTS

### Additional dependencies

```python

```

LGTM @jacobtylerwalls 

```python
(venv3131) markbyrne@Marks-MacBook-Air-2 programming % python -m pylint --version
pylint 4.0.0-dev0
astroid 3.3.6
Python 3.13.1 (v3.13.1:06714517797, Dec  3 2024, 14:00:22) [Clang 15.0.0 (clang-1500.3.9.4)]
(venv3131) markbyrne@Marks-MacBook-Air-2 programming % cat x.py
import collections.abc
(venv3131) markbyrne@Marks-MacBook-Air-2 programming % python -V
Python 3.13.1
(venv3131) markbyrne@Marks-MacBook-Air-2 programming % python -m pylint x.py
************* Module x
x.py:1:0: C0114: Missing module docstring (missing-module-docstring)
x.py:1:0: W0611: Unused import collections.abc (unused-import)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)

(venv3131) markbyrne@Marks-MacBook-Air-2 programming % 
```

reproducible on my end as well

```
import.py:1:0: E0401: Unable to import 'collections.abc' (import-error)
import.py:1:0: W0611: Unused import collections.abc (unused-import)
```

with the following:
```
pylint 4.0.0-dev0 (latest main)
astroid 3.3.5
Python 3.13.1
```

--- a/doc/whatsnew/3/3.3/index.rst
+++ b/doc/whatsnew/3/3.3/index.rst
@@ -14,6 +14,35 @@ Summary -- Release highlights
 
 .. towncrier release notes start
 
+What's new in Pylint 3.3.3?
+---------------------------
+Release date: 2024-12-23
+
+
+False Positives Fixed
+---------------------
+
+- Fix false positives for ``undefined-variable`` for classes using Python 3.12
+  generic type syntax.
+
+  Closes #9335 (`#9335 <https://github.com/pylint-dev/pylint/issues/9335>`_)
+
+- Fix a false positive for `use-implicit-booleaness-not-len`. No lint should be emitted for
+  generators (`len` is not defined for generators).
+
+  Refs #10100 (`#10100 <https://github.com/pylint-dev/pylint/issues/10100>`_)
+
+
+
+Other Bug Fixes
+---------------
+
+- Fix ``Unable to import 'collections.abc' (import-error)`` on Python 3.13.1.
+
+  Closes #10112 (`#10112 <https://github.com/pylint-dev/pylint/issues/10112>`_)
+
+
+
 What's new in Pylint 3.3.2?
 ---------------------------
 Release date: 2024-12-01

--- a/doc/whatsnew/fragments/10100.false_positive
+++ b/doc/whatsnew/fragments/10100.false_positive
@@ -1,4 +1,0 @@
-Fix a false positive for `use-implicit-booleaness-not-len`. No lint should be emitted for
-generators (`len` is not defined for generators).
-
-Refs #10100

--- a/doc/whatsnew/fragments/10112.bugfix
+++ b/doc/whatsnew/fragments/10112.bugfix
@@ -1,3 +1,0 @@
-Fix ``Unable to import 'collections.abc' (import-error)`` on Python 3.13.1.
-
-Closes #10112

--- a/doc/whatsnew/fragments/9335.false_positive
+++ b/doc/whatsnew/fragments/9335.false_positive
@@ -1,4 +1,0 @@
-Fix false positives for ``undefined-variable`` for classes using Python 3.12
-generic type syntax.
-
-Closes #9335

--- a/pylint/__pkginfo__.py
+++ b/pylint/__pkginfo__.py
@@ -9,7 +9,7 @@ It's updated via tbump, do not modify.
 
 from __future__ import annotations
 
-__version__ = "3.3.2"
+__version__ = "3.3.3"
 
 
 def get_numversion_from_version(v: str) -> tuple[int, int, int]:

--- a/tbump.toml
+++ b/tbump.toml
@@ -1,7 +1,7 @@
 github_url = "https://github.com/pylint-dev/pylint"
 
 [version]
-current = "3.3.2"
+current = "3.3.3"
 regex = '''
 ^(?P<major>0|[1-9]\d*)
 \.

--- a/towncrier.toml
+++ b/towncrier.toml
@@ -1,5 +1,5 @@
 [tool.towncrier]
-version = "3.3.2"
+version = "3.3.3"
 directory = "doc/whatsnew/fragments"
 filename = "doc/whatsnew/3/3.3/index.rst"
 template = "doc/whatsnew/fragments/_template.rst"

